src/map/zone_entities.cpp

```diff
@@ -1103,6 +1103,21 @@ CCharEntity* CZoneEntities::GetCharByID(uint32 id)
     return nullptr;
 }
 
+const NPCEntityList_t CZoneEntities::GetNPCsByName(std::string const& pattern)
```

Comment:
As Zach already mentioned, you could likely use `zone:queryEntitiesByName()` and not have to add any of the `GetNPCsByName()`. Less code is always better! An added benefit is that function uses memoization so the cost is essentially free on repeat calls. If you have a good reason to keep this function then I would suggest adding in memoization to significantly reduce the cost.

Reply:
I was looking into the method. What would be a good way to iterate through all zones in lua?

I was going to have a follow up MR to  refactor this `zone:queryEntitiesByName()` method into something that takes a builder object so that it can be used by other types of search (removing other search bindings and leaving only this one).
When I backstopped myself from doing the refactor, I decided to keep the code paths separate, but it did mean adding a few methods so I am happy to see how it would look like if I reused the existing one.

Reply:
Ah I think you meant having a binding that calls through to the method for each zone right?

Reply:
As discussed on discord, will try to create a new binding that uses this `zone:queryEntitiesByName`, as well as use pattern matching in `zone:queryEntitiesByName`